packages/back-end/src/api/sdk-payload/getSdkPayload.ts

```diff
@@ -25,11 +26,20 @@ export const getSdkPayload = createApiRequestHandler()(
       hashSecureAttributes,
     } = await getPayloadParamsFromApiKey(key, req);
 
+    const environmentDoc = req.context.org?.settings?.environments?.find(
+      (e) => e.id === environment
+    );
+    const filteredProjects = filterProjectsByEnvironment(
+      projects,
+      environmentDoc,
+      true
+    );
```

Comment:
What if an SDK Connection is filtered to just project A and it's for dev.  You remove project A from the dev environment.

So before filtering, projects = `["A"]`, but after it's an empty array.  Won't that change the behavior here?  Instead of only including features in project A, it now includes all features (no project scoping at all).

Reply:
Yes, this is the implicit danger in removing a project scope. It affects tied entities. In this case, there is no way to have a "zero project" SDK connection, so we default to the next best thing, back to all projects.

We warn before removing the project
> You have made the projects filter more restrictive than before. 2 SDK Connections using this environment may be impacted.

However, it's not exactly accurate per your example. Removing project A from the sdk connection resets it to [], which here is "less restrictive". Maybe we should call this out in the UI?

Reply:
how about this?
<img width="610" alt="image" src="https://github.com/growthbook/growthbook/assets/7927873/e0c3259e-1793-4373-b060-adc3fe936a4c">